src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig

```diff
@@ -201,6 +201,7 @@
         {%- for child in form %}
             {{- form_widget(child, {
                 parent_label_class: label_attr.class|default(''),
+                label_translation_parameters: choice_translation_parameters|default([]),
```

Comment:
Shouldn't this be `label_translation_parameters: choice_translation_parameters|default(label_translation_parameters)` for backwards compatibility (same everywhere else)?

Reply:
@xabbuh I really don't know.

The fact that `label_translation_parameters` is used for the choices is documented nowhere. I feel like it's working this way just by luck and I don't know if it was intending to be like this.

`translation_domain` is used by default, but it's a generic option.
`label_translation_parameters` is not a generic option, it seems really specific to the `label`.

For example, `choice_attr` doesn't seems to use `label_attr` as default value...

Reply:
I am pretty sure that quite some people will have figured out that they could use the `label_translation_parameters` option to achieve this. I'd rather not break that even though it was not intended to be used like this.

Reply:
> I am pretty sure that quite some people will have figured out that they could use the `label_translation_parameters` option to achieve this.

This behaviour wasn't intend to work like this and was not documented. It could be considered as a bug.

> I'd rather not break that even though it was not intended to be used like this.

`attr_translation_parameters` doesn't fallback on `label_translation_parameters`
`help_translation_parameters` doesn't fallback on `label_translation_parameters`

This would be inconsistent that
`choice_translation_parameters` fallback on `label_translation_parameters`

Plus, it would introduce a bug on this new feature.

With something like
```
# translations/messages.en.yaml
    form.order.yes: 'I confirm my order to the company'
    form.order.no: 'I cancel my order to the company'
    form.order.label: 'I cancel my order to company'
```
```
$builder->add('id', null, [
        'label' => 'form.order.label'
        'label_translation_parameters' => [
            'company' => 'ACME Inc.',
        ],
        'choice' => [
            'form.order.yes' => true,
            'form.order.no' => false,
        ],
        'choice_translation_parameters' => [],
    ]);
```

I would expect to read, 
```
    form.order.yes: 'I confirm my order to the company'
    form.order.no: 'I cancel my order to the company'
    form.order.label: 'I cancel my order to ACME inc'
```
But I will get
```
    form.order.yes: 'I confirm my order to the ACME inc'
    form.order.no: 'I cancel my order to the ACME inc'
    form.order.label: 'I cancel my order to ACME inc'
```

Reply:
@xabbuh WDYT about

The fact that
- attr_translation_parameters doesn't fallback on label_translation_parameters
- help_translation_parameters doesn't fallback on label_translation_parameters

And
```
$builder->add('id', null, [
        'label' => 'form.order.label'
        'label_translation_parameters' => [
            'company' => 'ACME Inc.',
        ],
        'choice' => [
            'form.order.yes' => true,
            'form.order.no' => false,
        ],
        'choice_translation_parameters' => [],
    ]);
```
will be buggy

Reply:
What do you mean with that your last example will be buggy?

Reply:
With something like
```
# translations/messages.en.yaml
    form.order.yes: 'Yes, I confirm my order to the company'
    form.order.no: 'No, I cancel my order to the company'
    form.order.label: 'Do you confirm your order to company ?'
```
```
$builder->add('id', null, [
        'label' => 'form.order.label'
        'label_translation_parameters' => [
            'company' => 'ACME Inc.',
        ],
        'choice' => [
            'form.order.yes' => true,
            'form.order.no' => false,
        ],
        'choice_translation_parameters' => [],
    ]);
```

I would expect to read, 
```
    form.order.yes: 'Yes, I confirm my order to the company'
    form.order.no: 'No, I cancel my order to the company'
    form.order.label: 'Do you confirm your order to ACME inc ?'
```
But if I fallback to `label_translation_parameters`, I will get
```
    form.order.yes: 'Yes, I confirm my order to the ACME inc'
    form.order.no: 'No, I cancel my order to the ACME inc'
    form.order.label: 'Do you confirm your order to ACME inc ?'
```

Reply:
Friendly ping, 
Did you miss my answer @xabbuh ? Is my example clear ?

Reply:
I see what you mean, but I don't believe that something like that is very common as you will usually don't have placeholders like a plain `company` string, but `%company or `{company}` or the like instead.

And from my experience, people discover all these silently "working" things any there will be complaints if we break that now.

Reply:
`choice_attr` does not default on `label_attr`.
`choice_translation_domain` does not default on `translation_domain`.
`choice_translation_parameters` will default 'label_translation_parameters`.

Best way to lose consistency and lose developper when they use this option.

> And from my experience, people discover all these silently "working" things any there will be complaints if we break that now.

Working and break are opinionated words here.
![xkcd workflow](https://imgs.xkcd.com/comics/workflow.png)

Currently the fact that `label_translation_parameters` are used is not even consistent:
In the two following example, the parameters was currently `{}` which does mean that there is no use of `label_translation_parameters` for a select.
https://github.com/symfony/symfony/pull/36851/files#diff-ebe082b8f8e40b852f895d7712b30730R83
https://github.com/symfony/symfony/pull/36851/files#diff-ebe082b8f8e40b852f895d7712b30730R88

So If I want to avoid a opinionated BC-break, I just have to default to `label_translation_parameters` for half the case.

How are introduced BC-break in Symfony ?
Because this would need to be changed in Symfony 6.0 and master is the only available branch.

Reply:
Hi @xabbuh @nicolas-grekas

How does BC-break are introduce in symfony ?

Let's say I use `label_translation_parameters` as default value for `choice_translation_parameters` in order to be fully BC (even if the previous behavior wasn't consistent and wasn't documented).

I would like that the `choice_translation_parameters` doesn't default on `label_translation_parameters` in the next major. In order to be consistent with `attr_translation_parameters` and `help_translation_parameters`.

What should I do ? 
I can't make the PR on the 6.0 branch, since it doesn't exist.
Should I add a comment `// NEXT_MAJOR: REMOVE THE DEFAULT VALUE` ?

Reply:
The process would be to trigger a deprecation when the fallback becomes active. But I honestly do not have a good idea how to implement that as we do not know whether one of the fallbacks will actually be used.

Reply:
So I don't see better solution than a `NEXT_MAJOR` comment... I pushed a new commit.
Maybe creating an issue with the 6.0 milestone ?

Reply:
We should not silently change the behaviour in 6.0. So that doesn't look like a valid option to me.

Reply:
The only option I see would be throwing a deprecation if `choice_translation_parameters` is `[]` and `label_translation_parameters` is not `[]`. With a message like
> The `choice_translation_parameters` won't fallback on `label_translation_parameters` on Symfony 6.0. Please check that it will not affect you or add the `choice_translation_parameters` option to your form. 

The only defect is that you can't get rid of every deprecation if you use `label_translation_parameters` and you don't need the `choice_translation_parameters` option...